core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -291,6 +291,20 @@ const CudaInternalDevices &CudaInternalDevices::singleton() {
 
 }  // namespace
 
+int CudaInternal::m_cudaDev;
+int CudaInternal::m_cudaArch;
+unsigned CudaInternal::m_multiProcCount;
+unsigned CudaInternal::m_maxWarpCount;
+std::array<CudaInternal::size_type, 3> CudaInternal::m_maxBlock;
+unsigned CudaInternal::m_maxSharedWords;
+uint32_t CudaInternal::m_maxConcurrency;
+int CudaInternal::m_shmemPerSM;
+int CudaInternal::m_maxShmemPerBlock;
+int CudaInternal::m_maxBlocksPerSM;
+int CudaInternal::m_maxThreadsPerSM;
+int CudaInternal::m_maxThreadsPerBlock;
+cudaDeviceProp CudaInternal::m_deviceProp;
+CudaInternal::size_type CudaInternal::m_scratchUnifiedSupported;
```

Comment:
Same comment as in #5444 consider declaring these `inline`

Reply:
Would you mind elaborating on the syntax you are proposing? 
Are you saying you just want to declare them as, e.g.,  `inline static int m_cudaDev;` and don't change anything else?

Reply:
Initialize them in the body of the class

Reply:
Daniel pointed out to me that this would lead to a bug because it would have different translation units

Reply:
Even then, I would only use `inline static` if I can provide a value that won't change (but the type doesn't allow `constexpr`).